From 9d7d296ea2b0390173265e128c67858d730c4877 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Fri, 6 Nov 2020 15:22:40 -0500
Subject: [PATCH] SurfaceFlinger: Don't cleanup resources from previous frame

Causes bad lag on some legacy devices.

Change-Id: I89913d214c7377c73bd307696dbf9aac2f9a5c0a
(cherry picked from commit 4e0058d58f2c374be5a274b17a41ada310ccf87f)
---
 services/surfaceflinger/Android.bp         | 1 +
 services/surfaceflinger/SurfaceFlinger.cpp | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/services/surfaceflinger/Android.bp b/services/surfaceflinger/Android.bp
index c43a3c39e8..9f6bfcdd45 100644
--- a/services/surfaceflinger/Android.bp
+++ b/services/surfaceflinger/Android.bp
@@ -24,6 +24,7 @@ cc_defaults {
 cc_defaults {
     name: "libsurfaceflinger_defaults",
     defaults: [
+        "disable_postrender_cleanup_defaults",
         "surfaceflinger_defaults",
         "skia_renderengine_deps",
         "surfaceflinger_qcom_ext_defaults",
diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 4e46f27e75..65d459c874 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -2441,8 +2441,10 @@ void SurfaceFlinger::postComposition() {
     }
     getBE().mLastSwapTime = currentTime;
 
+#ifndef DISABLE_POSTRENDER_CLEANUP
     // Cleanup any outstanding resources due to rendering a prior frame.
     getRenderEngine().cleanupPostRender();
+#endif
 
     {
         std::lock_guard lock(mTexturePoolMutex);
